### ISSUE TITLE
Issue with shared preferences || Nav destination is unknown to navController

### ISSUE THREAD

**Issue by kriticalflare** [Zhuinden/Jetpack-Navigation-Fragment-NavGraph-Dagger-SavedStateHandle-FTUE-Experiment#4]
On finishing the registration flow , i have noticed that the username field would often not get saved to shared prefs ( verified by pulling the shared prefs xml from device)  ie Empty map.
When the username field did get _saved_ , the app would crash on launch with following crash log 

OS: Android 10
App built from master @ https://github.com/Zhuinden/jetpack-navigation-ftue-sample/commit/b3fbbf7929e3e1525774d796bbb42a4f231276b3

```


time: 1591299111785
msg: java.lang.IllegalArgumentException: navigation destination com.zhuinden.jetpacknavigationfragmentnavgraphdaggersavedstatehandleftueexperiment:id/logged_out_to_logged_in is unknown to this NavController
stacktrace: java.lang.IllegalArgumentException: navigation destination com.zhuinden.jetpacknavigationfragmentnavgraphdaggersavedstatehandleftueexperiment:id/logged_out_to_logged_in is unknown to this NavController
	at androidx.navigation.NavController.navigate(NavController.java:865)
	at androidx.navigation.NavController.navigate(NavController.java:806)
	at androidx.navigation.NavController.navigate(NavController.java:792)
	at androidx.navigation.NavController.navigate(NavController.java:987)
	at com.zhuinden.jetpacknavigationdaggersavedstatehandleftueexperiment.features.splash.SplashFragment$finishSplash$1.run(SplashFragment.kt:35)
	at android.os.Handler.handleCallback(Handler.java:883)
	at android.os.Handler.dispatchMessage(Handler.java:100)
	at android.os.Looper.loop(Looper.java:214)
	at android.app.ActivityThread.main(ActivityThread.java:7356)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:491)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:915)

```

**Comment by Zhuinden** [comment]
Thanks for the report, it's most likely due to the newly introduced argument passing example.

Odd because I thought I checked all cases but I guess one navigation wasn't replaced to use NavDirections. I'll look at it soon

**Comment by Zhuinden** [comment]
Repro: after login, terminate the app while it is in foreground.

**Comment by Zhuinden** [comment]
I didn't want the `argument passing example` commit to contain bugs, so I did a force push. Apparently there was a typo on that SplashGraph used the NavDirections of LoggedOutGraph by accident, as each of them has their own actions, and I guess I mixed them up.

I'm not sure about the shared pref issue, I can't seem to repro that. Thanks for the report.

https://github.com/Zhuinden/jetpack-navigation-ftue-sample/commit/3ffdd6d695471ca7b2d99d510fd146f74eabce84#diff-0757ddad5303a6f8c2c1ff42f9e14964R35
